### PR TITLE
MINOR: Add Kafka 3.5 documentation and javadoc

### PR DESCRIPTION
**Pull request by mimaison**
Generated while running the 3.5.0 RC0 build

### CODE REVIEW

**Review by bbejeck — APPROVED** [review]

File: 35/documentation.html [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    <div class="toc-handle">&lt;</div>
+  </div>
+  <div class="docs-nav">
+    <!--#include virtual="toc.html" -->
+  </div>
+  <div class="right">
+    <!--//#include virtual="../includes/_docs_banner.htm" -->
+    
+    <h1>Documentation</h1>
+    <h3>Kafka 3.4 Documentation</h3>
```
Comment (jlprat) [inline 35/documentation.html:36]: Shouldn't this read 3.5?

File: 35/documentation.html [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                    <a href="/23/documentation.html">2.3.X</a>, 
+                    <a href="/24/documentation.html">2.4.X</a>, 
+                    <a href="/25/documentation.html">2.5.X</a>, 
+                    <a href="/26/documentation.html">2.6.X</a>, 
+                    <a href="/27/documentation.html">2.7.X</a>,
+                    <a href="/28/documentation.html">2.8.X</a>,
+                    <a href="/30/documentation.html">3.0.X</a>.
+                    <a href="/31/documentation.html">3.1.X</a>.
+                    <a href="/32/documentation.html">3.2.X</a>.
+                    <a href="/33/documentation.html">3.3.X</a>.
```
Comment (jlprat) [inline 35/documentation.html:60]: And here a link to 3.4 release

**Review by jlprat — COMMENTED** [review]
@mimaison I think the pages need a couple of updates. The documentation refers to itself as 3.4 instead of 3.5 and there is no link to 3.4 docs.

File: 35/documentation.html (hunk 1 above)
Comment (mimaison) [reply 35/documentation.html:36]: Good catch! I opened https://github.com/apache/kafka-site/pull/514

**Pull request merged**